File: cpp/log-to-console/src/LogToConsole.cpp
/*
 * (c) 2018-2020 Ionic Security Inc.
 * By using this code, I agree to the Terms & Conditions (https://dev.ionic.com/use.html)
 * and the Privacy Policy (https://www.ionic.com/privacy-notice/).
 */

#include "ISLog.h"
#include <stdio.h>
#include <cstdlib>
#include <iostream>

int main(int argc, char* argv[]) {

    std::string sLogChannel = "ionic-cpp-sample";

    // initialize logger
    ISLogFilterSeverity * pFilter = new ISLogFilterSeverity(SEV_TRACE); 
    ISLogWriterConsole * pConsoleWriter = new ISLogWriterConsole();
    pConsoleWriter->setFilter(pFilter);
    ISLogSink * pSink = new ISLogSink();
    pSink->registerChannelName(sLogChannel);
    pSink->registerWriter(pConsoleWriter);
    ISLogImpl * pLogger = new ISLogImpl(true);
    pLogger->registerSink(pSink);
    ISLog::setSingleton(pLogger);

    // write log
    ISLOG_INFO(sLogChannel.c_str(), "Sample log entry");
}

File: java/log-to-console/src/main/java/com/ionic/samples/LogToConsole.java
/*
 * (c) 2018-2020 Ionic Security Inc.
 * By using this code, I agree to the Terms & Conditions (https://dev.ionic.com/use.html)
 * and the Privacy Policy (https://www.ionic.com/privacy-notice/).
 * 
 * Developed with Ionic Java SDK 2.1.0
 */

package com.ionic.samples;

import java.util.logging.Level;
import java.util.logging.Logger;
import java.util.logging.ConsoleHandler;

public class LogToConsole
{
    public static void main(String[] args)
    {
        String sLogChannel = "com.ionic.sdk.sampleapp";

        // initialize logger
        final Logger logger = Logger.getLogger(sLogChannel);
        ConsoleHandler ch = new ConsoleHandler();
        logger.addHandler(ch);
        logger.setLevel(Level.INFO);

        // write log 
        logger.info("Sample log entry");
    }
}
